### ISSUE TITLE
Ensure all LibGit2SharpException derived types exposes a string.Format friendly ctor

### ISSUE THREAD

**Issue by nulltoken** [libgit2/libgit2sharp#1199]
For each LibGit2SharpException derived type, find its usage and when a construct similar to the following one exist...

```
if (res == (int)GitErrorCode.Exists)
{
    throw new EntryExistsException(
        String.Format("A custom transport for '{0}' is already registered", prefix));
}
```

...add a new ctor to the type and update the calling code to:

```
if (res == (int)GitErrorCode.Exists)
{
    throw new EntryExistsException(CultureInfo.InvariantCulture,
        "A custom transport for '{0}' is already registered", prefix);
}
```

cf  #1095, #1198

/cc @rooflz

**Comment by whoisj** [comment]
:+1: Great idea.

**Comment by odedw** [comment]
I'd like to take a look at this.

**Comment by nulltoken** [comment]
@odedw :rocket:

**Comment by odedw** [comment]
Pull request #1202

**Comment by nulltoken** [comment]
Fixed by #1202
